File: Motocitizen/src/motocitizen/app/general/popups/MessagesPopup.java
package motocitizen.app.general.popups;

import android.widget.PopupWindow;

import motocitizen.accident.Message;
import motocitizen.content.Content;
import motocitizen.utils.MyUtils;

public class MessagesPopup extends PopupWindowGeneral {
    private final Message message;

    public MessagesPopup(int id, int accId) {
        super();
        message = Content.getInstance().get(accId).getMessages().get(id);
    }

    public PopupWindow getPopupWindow() {
        content.addView(copyButtonRow(message.getOwner() + ": " + message.getText()), layoutParams);
        for (String phone : MyUtils.getPhonesFromText(message.getText())) {
            content.addView(phoneButtonRow(phone), layoutParams);
        }
        popupWindow.setContentView(content);
        return popupWindow;
    }
}

File: Motocitizen/src/motocitizen/ui/popups/MessagesPopup.kt
package motocitizen.ui.popups

import android.content.Context

import motocitizen.content.Content
import motocitizen.content.message.Message

import motocitizen.utils.getPhonesFromText

class MessagesPopup(context: Context, id: Int) : PopupWindowGeneral(context) {
    private val message: Message = Content.message(id)

    init {
        rootView.addView(copyButtonView(message.owner.toString() + ": " + message.text), layoutParams)
        getPhonesFromText(message.text).forEach { phone -> rootView.addView(phoneButtonView(phone), layoutParams) }
        contentView = rootView
    }
}
